### PR TITLE
Ide gripes

### PR DESCRIPTION
**Pull request by alan-stokes**
fixes #

Things to do in this repository
===============================
- [ ] javadoc correct.
- [ ] test case(s).

### CODE REVIEW

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Router.java [hunk 1]
```diff
@@ -16,12 +16,7 @@
  */
 package uk.ac.manchester.spinnaker.machine;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.EnumMap;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (dkfellows) [inline SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Router.java]: Style rules reject this change. List everything explicitly.

Comment (alan-stokes) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Router.java]: the fact that the ide put that in shocks me to the core. IDE BADDNESS!!! will fix once I've got to the end of the utils

Comment (alan-stokes) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Router.java]: should be fixed now. it is def a automatic thing from the ide, had to keep replacing it. lol. am looking at how to switch it off now

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Machine.java [hunk 2]
```diff
@@ -753,17 +745,11 @@ private void addFpgaLinks(int rootX, int rootY, InetAddress address) {
                         fpgaEnum.id, fpgaEnum.fpgaId, location,
                         fpgaEnum.direction, address);
                 Map<FpgaId, Map<Integer, FPGALinkData>> byAddress;
-                byAddress = fpgaLinks.get(address);
-                if (byAddress == null) {
-                    byAddress = new HashMap<>();
-                    fpgaLinks.put(address, byAddress);
-                }
+                byAddress = fpgaLinks.computeIfAbsent(
```
Comment (Christian-B) [inline SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/Machine.java]: This must be an IDE suggestion.   
No way the Alan we know would come up with this change by himself.
Surprised he agree to it even with IDE help.
By the way the IDE is right.

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/MachineVersion.java [hunk 3]
```diff
@@ -210,7 +210,7 @@
      */
     public static MachineVersion byId(Integer id) {
         for (MachineVersion possible: MachineVersion.values()) {
-            if (possible.id == id) {
+            if (possible.id.equals(id)) {
```
[the remark this answers is not in the document]
Comment (Christian-B) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/MachineVersion.java]: possible.id.equals(id) IS NOT valid as many versions ie all multiple board ones have a null id,

[the remark this answers is not in the document]
Comment (alan-stokes) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/MachineVersion.java]: fair enough, that was ide recommandation.......but integer can be none, so shouldnt values include it?

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/bean/ChipBean.java [hunk 4]
```diff
@@ -52,7 +52,7 @@ public ChipBean(@JsonProperty(value = "x", required = true) int x,
             @JsonProperty(value = "y", required = true) int y,
             @JsonProperty(value = "details", required = true)
             ChipDetails details,
-            @JsonProperty(value = "resources", required = false)
+            @JsonProperty(value = "resources")
```
[the remark this answers is not in the document]
Comment (alan-stokes) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/bean/ChipBean.java]: im going to just accept you there. as i dont even remeber putting it in

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/SpiNNakerTriadGeometry.java [hunk 5]
```diff
@@ -90,8 +90,8 @@ private SpiNNakerTriadGeometry(int triadHeight, int triadWidth,
                         new Location(root.getX(), root.getY() - triadWidth));
             }
         }
-        this.xCenterer = xCenterer;
-        this.yCenterer = yCenterer;
+        this.xCenterer = xCenter;
+        this.yCenterer = yCenter;
```
[the remark this answers is not in the document]
Comment (alan-stokes) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/SpiNNakerTriadGeometry.java]: lol, i didnt even see it. yea go ahead and fiddle. im deep in c code at moment. plans was to catch up on all prs tomorrow if i can get this to compile by end of day

File: SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/MachineVersion.java (hunk 3 above)
Comment (Christian-B) [reply SpiNNaker-machine/src/main/java/uk/ac/manchester/spinnaker/machine/MachineVersion.java]: After testing and discussion with @dkfellows we decided the safest solution is:

    public static MachineVersion byId(Integer id) {
        for (MachineVersion possible: MachineVersion.values()) {
            if (possible.id != null && possible.id.equals(id)) {
                return possible;
            }
        }
        throw new IllegalArgumentException("No Board version with id: " + id);
    }

**Comment by Christian-B** [comment]
Ok to merge when Green!

**Pull request merged**